lib/iris/cube.py

```diff
@@ -4020,17 +4029,24 @@ def aggregated_by(self, coords, aggregator, **kwargs):
             dimensions=dimension_to_groupby, dim_coords=True
         ) or [None]
         for coord in groupby.coords:
+            new_coord = coord.copy()
+
+            try:
+                if climatological:
+                    new_coord.climatological = True
```

Comment:
I thought it looked like the coord needed to be copied and then this added to it because otherwise we were editing inplace, but I see now that it's already been copied by the call to `iris.util._strip_metadata_from_dims` so I think you're right that this can all move/go

Reply:
Moved

Reply:
Turns out I can't do this as-is because the `self.coord_dims(coord)` call on line 4041 fails if the metadata has changed (including climatologicalness). I can hack it back long enough to do the check, or add the climatological after the copy, or carry the coord_dims info from earlier or do something more involved.

Reply:
Unfortunately yes, you can have coords with the same name.  Sorry I may have led you on a wild goose chase here and maybe your original approach was the simplest.  Or maybe @bjlittle may have a better suggestion as he is the Metadata King 🤴

Reply:
@rcomer has just told me that's not a good assumption, I'll find another way

Reply:
I'm now maintaining a lookup on the `_Groupby` object that records which cube coord the resulting coords came from